acme/src/main/java/io/micronaut/configuration/acme/background/AcmeCertRefresherTask.java

```diff
@@ -40,64 +45,58 @@
 
     private static final Logger LOG = LoggerFactory.getLogger(AcmeCertRefresherTask.class);
 
-    @Value("${micronaut.ssl.acme.renew.within:30}")
-    private int renewWithinDays = 30;
-
-    @Value("${micronaut.ssl.acme.tos.agree:false}")
-    private boolean agreeToTOS = false;
-
-    @Property(name = "micronaut.ssl.acme.domain")
-    private String domain;
-
     private AcmeService acmeService;
+    private final AcmeConfiguration acmeConfiguration;
 
     /**
      * Constructs a new Acme cert refresher background task.
      *
      * @param acmeService Acme service
      */
-    @Inject
-    public AcmeCertRefresherTask(AcmeService acmeService) {
+    public AcmeCertRefresherTask(AcmeService acmeService, AcmeConfiguration acmeConfiguration) {
```

Comment:
The unit tests now complain about this constructor change so those tests will need to be updated as well.

Reply:
I setup TravisCI so if you rebase with master the tests should be in a better state for you. There were also some issues with `java.nio.channels.UnresolvedAddressException` that you might run into due to the way I was originally doing the hostname but those should be fixed and not require any custom configuration hopefully.